src/hotspot/share/code/nmethod.cpp

```diff
@@ -2390,6 +2391,8 @@ class IsUnloadingState: public AllStatic {
 };
 
 bool nmethod::is_unloading() {
+  MACOS_AARCH64_ONLY(os::thread_wx_enable_write());
```

Comment:
This is very opaque. Why is the write enable placed here rather than lower down where we drop past the return true and return false cases and potentially start monkeying around with state updates?

Currently, it looks as if the fact that a caller has tested the state may imply that said caller might write to the nmethod even when we determine the state via the first two checks. Is that really true? If so then an explanation as to why would be highly appropriate.